src/Word/DashToSeparator.php

```diff
@@ -14,15 +17,42 @@
  *     ...
  * }
  * @template TOptions of Options
- * @extends AbstractSeparator<TOptions>
+ * @implements FilterInterface<mixed>
  */
-final class DashToSeparator extends AbstractSeparator
+final class DashToSeparator implements FilterInterface
 {
+    protected string $separator = ' ';
+
+    /**
+     * @param Options|string $separator Space by default
+     */
+    public function __construct(string|array $separator = ' ')
+    {
+        if (is_array($separator) && isset($separator['separator'])) {
+            $this->setSeparator($separator['separator']);
+
+            return;
+        }
+
+        $this->setSeparator($this->separator);
+    }
+
+    public function __invoke(mixed $value): mixed
+    {
+        return $this->filter($value);
+    }
+
     public function filter(mixed $value): mixed
     {
-        return self::applyFilterOnlyToStringableValuesAndStringableArrayValues(
-            $value,
-            Closure::fromCallable([$this, 'filterNormalizedValue'])
+        if (! is_array($value)) {
+            if (! is_scalar($value)) {
+                return $value;
+            }
+            return $this->filterNormalizedValue((string) $value);
+        }
+
+        return $this->filterNormalizedValue(
+            array_map(static fn($item) => is_scalar($item) ? (string) $item : $item, $value)
```

Comment:
While this causes a static analysis issue, I believe it is consistent with what is being done in AbstractFilter::applyFilterOnlyToStringableValuesAndStringableArrayValues . So changing it would likely introduce a change in behaviour